tests/sandbox-retry-behavior.test.ts

```diff
@@ -0,0 +1,94 @@
+import { describe, it, expect, beforeEach, afterEach } from 'vitest'
+import nock from 'nock'
+import { CodeSandbox } from '../src/index'
+import { 
+  mockForkSandboxSuccess, 
+  mockStartVMSuccess, 
+  mockStartVMFailure, 
+  setupTestEnvironment, 
+  cleanupTestEnvironment 
+} from './test-utils'
+
+describe('Create operation retry behavior', () => {
+  beforeEach(() => {
+    setupTestEnvironment()
+  })
+
+  afterEach(() => {
+    cleanupTestEnvironment()
+  })
+
+  it('should fail immediately on fork API error (no retry for fork)', async () => {
+    // Mock fork to fail once - should fail immediately since fork doesn't retry
+    const forkScope = nock('https://api.codesandbox.io')
+      .post('/sandbox/pcz35m/fork')
+      .reply(500, { error: { errors: ['Fork failed'] } })
+
+    const sdk = new CodeSandbox()
+    
+    // Should fail immediately without retries
+    const startTime = Date.now()
+    await expect(sdk.sandboxes.create()).rejects.toThrow()
+    const duration = Date.now() - startTime
+    
+    // Should fail quickly (no retry delays)
+    expect(duration).toBeLessThan(1000)
+    expect(forkScope.isDone()).toBe(true)
+  })
+
+  it('should retry start VM failures and eventually succeed', async () => {
+    // Mock successful fork
+    const forkScope = mockForkSandboxSuccess('test-sandbox-start-retry')
+
+    // Mock start VM to fail twice, then succeed on 3rd attempt
+    mockStartVMFailure(2, 'Start failed')
+    const startScope = mockStartVMSuccess('test-sandbox-start-retry')
+
+    const sdk = new CodeSandbox()
+    
+    // Should succeed after start VM retries
+    const sandbox = await sdk.sandboxes.create()
```

Comment:
Should we confirm the retries count, is there a way to do that? Or do you think it's useless because we're mocking things anyway?

Reply:
Good point, let me update 😄